larpandoracontent/LArTwoDReco/LArCosmicRay/TrackExtensionRefinementAlgorithm.cc

```diff
@@ -0,0 +1,416 @@
+                {
+                    // ATTN: To avoid counting same hit twice
+                    const ClusterToCaloHitListMap::iterator iter(clusterToCaloHitListMap.find(pCluster));
+                    if (iter != clusterToCaloHitListMap.end())
+                    {
+                        if (std::find(iter->second.begin(), iter->second.end(), pCaloHit) != iter->second.end())
+                            continue;
+                    }
+
+                    CartesianVector hitPosition(m_hitWidthMode ?
+                        LArHitWidthHelper::GetClosestPointToLine2D(extrapolatedStartPosition, extrapolatedDirection, pCaloHit) :
+                        pCaloHit->GetPositionVector());
+                    
+                    if (!IsInLineSegment(extrapolatedStartPosition, extrapolatedEndPosition, hitPosition))
+                        continue;
+
+                    const float transverseDistanceFromLine(extrapolatedDirection.GetCrossProduct(hitPosition - extrapolatedStartPosition).GetMagnitude());
+                    
+                    if (transverseDistanceFromLine > m_distanceToLine)
+                        continue;
+
+                    ++hitsCollected;
+                    
+                    runningFitPositionVector.push_back(hitPosition);
+                    clusterToCaloHitListMap[pCluster].push_back(pCaloHit);
+                }
+            }
+        }
+        catch (const StatusCodeException &)
+        {
+            return;
+        }
+
+        ++count;
+    }
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TrackExtensionRefinementAlgorithm::AreExtrapolatedHitsGood(const ClusterToCaloHitListMap &clusterToCaloHitListMap,
+    const bool isHigherXBoundary, ClusterEndpointAssociation &clusterAssociation) const
+{
+    CaloHitVector extrapolatedHitVector;
+    for (const auto &entry : clusterToCaloHitListMap)
+        extrapolatedHitVector.insert(extrapolatedHitVector.begin(), entry.second.begin(), entry.second.end());
+
+    // ATTN: Extrapolated hit checks require extrapolatedHitVector to be ordered from upstream -> downstream merge point  
+    std::sort(extrapolatedHitVector.begin(), extrapolatedHitVector.end(), SortByDistanceAlongLine(clusterAssociation.GetUpstreamMergePoint(),
+        clusterAssociation.GetConnectingLineDirection(), m_hitWidthMode));
+    
+    if (!this->IsExtrapolatedEndpointNearBoundary(extrapolatedHitVector, isHigherXBoundary, clusterAssociation))
+        return false;
+
+    if (clusterToCaloHitListMap.empty())
+        return true;
+
+    // ISOBEL - THIS COULD BE USEFUL IF GAP SIZE IS SMALL
+    if (!this->IsTrackContinuous(clusterAssociation, extrapolatedHitVector))
+    {
+        std::cout << "GAP IN HIT VECTOR" << std::endl;
+        PandoraMonitoringApi::ViewEvent(this->GetPandora());
+        return false;
+    }
```

Comment:
Are the comment and cout both still needed?

Reply:
Sorry no, i must have forgotten to remove the prints. It should read

if (!this->IsTrackContinuous(clusterAssociation, extrapolatedHitVector))
        return false;